src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQueryDatasetPdao.java

```diff
@@ -1428,4 +1430,35 @@ private static String bqStringValue(FieldValueList fieldValue, String fieldName)
     }
     return null;
   }
+
+  /// Snapshot Builder
+  // Pass in SQL query and map into generic BigQuery TableResult
+  public <T> List<T> runSnapshotBuilderQuery(
+      String sql, Dataset dataset, Function<TableResult, List<T>> aggregateSnapshotBuilderResults) {
+    try {
+      final BigQueryProject bigQueryProject = BigQueryProject.from(dataset);
+      final TableResult result = bigQueryProject.query(sql);
+      return aggregateSnapshotBuilderResults.apply(result);
+    } catch (InterruptedException ex) {
+      throw new PdaoException("Snapshot builder query was interrupted", ex);
+    }
+  }
+
+  // Convert BigQuery TableResult into List<SnapshotBuilderConcept>
+  public Function<TableResult, List<SnapshotBuilderConcept>>
+      aggregateSnapshotBuilderConceptResults() {
+    return (result) -> {
+      final List<SnapshotBuilderConcept> concepts = new ArrayList<>();
+      result
+          .iterateAll()
+          .forEach(
+              rows -> {
+                SnapshotBuilderConcept concept = new SnapshotBuilderConcept();
+                concept.id((int) (rows.get("concept_id").getLongValue()));
+                concept.name(rows.get("concept_name").getStringValue());
+                concepts.add(concept);
+              });
+      return concepts;
+    };
```

Comment:
Instead of a function that returns a function, you can write the function that's being used directly. Also you can use a spliterator here to stream the result. Here's both together:
```suggestion
  public static List<SnapshotBuilderConcept> aggregateSnapshotBuilderConceptResults(TableResult result) {
    return StreamSupport.stream(result.iterateAll().spliterator(), false)
        .map(
            row ->
                new SnapshotBuilderConcept()
                    .id((int) (row.get("concept_id").getLongValue()))
                    .name(row.get("concept_name").getStringValue()))
        .toList();
  }
```

And the call site change:
```
    List<SnapshotBuilderConcept> concepts =
        runSnapshotBuilderQuery(
            cloudSpecificSQL,
            dataset,
            BigQueryDatasetPdao::aggregateSnapshotBuilderConceptResults,
            azureSynapsePdao.aggregateSnapshotBuilderConceptResult());
```